Doc/reference/datamodel.rst

```diff
@@ -2215,23 +2215,80 @@ case the instance is itself a class.
 Emulating generic types
 -----------------------
 
-One can implement the generic class syntax as specified by :pep:`484`
-(for example ``List[int]``) by defining a special method:
+When using :term:`type annotations<annotation>`, it is often useful to
+*parameterize* a generic type using Python's square-brackets notation.
+For example, the annotation ``list[int]`` might be used to signify a
+:class:`list` in which all the elements are of type :class:`int`.
+
+.. seealso::
+
+   :pep:`484` - Type Hints
+      Introducing Python's framework for type annotations
+
+   :ref:`Generic Alias Types<types-genericalias>`
+      Documentation for objects representing parameterized generic classes
+
+   :class:`typing.Generic`
+      Inherit from :class:`typing.Generic` to implement generic classes that
+      can be parameterized at runtime and understood by static type-checkers.
+
+A class can generally only be parameterized if it defines the special
+classmethod ``__class_getitem__()``.
 
 .. classmethod:: object.__class_getitem__(cls, key)
 
    Return an object representing the specialization of a generic class
    by type arguments found in *key*.
 
-This method is looked up on the class object itself, and when defined in
-the class body, this method is implicitly a class method.  Note, this
-mechanism is primarily reserved for use with static type hints, other usage
-is discouraged.
+``__getitem__`` *versus* ``__class_getitem__``
```

Comment:
The formatting/indentation for this looks a little strange. 

![image](https://user-images.githubusercontent.com/28750310/140267976-b961d56b-3ce3-4ad5-97e0-1fdb8a1df1af.png)


Perhaps making this a subtitle would be better (you need to check what character that uses). So maybe like 
```
``__getitem__`` *versus* ``__class_getitem__``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
or using `~` or anything else, as long as it's used consistently for that subtitling level throughout the rest of the datamodel docs.